search-mapper/src/main/java/org/infinispan/search/mapper/work/impl/SearchIndexingPlanImpl.java

```diff
@@ -0,0 +1,124 @@
+package org.infinispan.search.mapper.work.impl;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.hibernate.search.mapper.pojo.model.spi.PojoRawTypeIdentifier;
+import org.hibernate.search.mapper.pojo.model.spi.PojoRuntimeIntrospector;
+import org.hibernate.search.mapper.pojo.work.spi.PojoIndexingPlan;
+import org.hibernate.search.util.common.impl.Throwables;
+import org.infinispan.search.mapper.mapping.EntityConverter;
+import org.infinispan.search.mapper.work.SearchIndexingPlan;
+
+public class SearchIndexingPlanImpl implements SearchIndexingPlan {
```

Comment:
Good point.
I applied the same pattern that we have for indexing Hibernate ORM entities, but, you're definitely right, the interceptor here reacts on single entity change, not to a series of changes that are bound to a transaction/flush as it is for Hibernate ORM.
If they changed the `QueryInterceptor`, so that it reacts on a series of changes, they probably should return to using an indexing plan.
Anyway, since keeping the indexing plan just for this could be a premature optimization, I would agree to use an indexer, as we did for mass indexers, that is easier and less resource expensive.
I’m taking note of the task, so that I will do the change after I restore the broadcast query and all the test cases.

Reply:
Created https://hibernate.atlassian.net/browse/HSEARCH-3902 to address the changes in PojoIndexer

Reply:
it sound great! thanks... it looks like a big performance improvement